pyproject.toml

```diff
@@ -28,7 +28,7 @@ dependencies = [
     "more_itertools~=9.0",
     "omegaconf~=2.3",
     "parse~=1.19.0",
-    "pip-tools>=6.5,<8.0",
+    "pip-tools>=7.0,<8.0",  # pip==23.2 is a breaking change that requires pip-tools>=7.0.0
```

Comment:
I don't think we should touch the lower bound. After all, there might be environments with old `pip` versions, right?

pip-tools already takes care of pinning pip appropriately

https://github.com/jazzband/pip-tools/blob/963642811c93a32cdaf730e8226d0fe091391882/pyproject.toml#L41